data/migrations/dba_operator.py

```diff
@@ -0,0 +1,142 @@
+import yaml
+import logging
+
+from functools import partial, wraps
+from alembic.operations import ops
+
+import sqlalchemy as sa
+
+
+logger = logging.getLogger(__name__)
+
+
+class Migration(object):
+    def __init__(self):
+        self._schema_hints = []
+
+    def dump_yaml_and_reset(self, stream, revision, down_revision):
+        migration_object = {
+            "apiVersion": "dbaoperator.app-sre.redhat.com/v1alpha1",
+            "kind": "DatabaseMigration",
+            "metadata": {"name": revision,},
+            "spec": {
+                "previous": down_revision,
+                "migrationContainerSpec": {
+                    "name": revision,
+                    "image": "quay.io/quaymigrations/realtest",
+                    "command": ["/quay-registry/quay-entrypoint.sh", "migrate", revision,],
+                },
+                "schemaHints": self._schema_hints,
+            },
+        }
+        yaml.dump(migration_object, stream)
+        self._schema_hints = []
+
+    def add_hints_from_ops(self, directive):
+        for op in directive.ops:
+            if isinstance(op, ops.ModifyTableOps):
```

Comment:
I wonder if we could make this easier to read via some sort of table of lambdas... not sure

Reply:
`if` statements are the worst, amirite?